### PR TITLE
Bug/729 add remove variables and components from schema for device model

### PR DESCRIPTION
**Pull request by maaikez**
## Describe your changes

With this pull request, it is possible to remove and add variables (and their attributes and characteristics) and components for all other components than only EVSE and Connector. 

It also removed the use of config.json, so everything is now added to the component schema's, which are renamed to component config. The values from the 'default config' are now added to the new 'component config' json files. 

There is one TODO left in the ocpp_types file. If the changes are ok there, I can also change the generator to do exactly that. Otherwise I have to solve it another way. 

I also made a pull request in everest core (https://github.com/EVerest/everest-core/pull/822) and are still busy with some changes in everest utils (https://github.com/EVerest/everest-utils/tree/feature/715-combine-component-schemas-and-config-in-one-file).

## Issue ticket number and link
#729 
#715 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: lib/ocpp/v201/init_device_model_db.cpp [hunk 1]
```diff
@@ -570,20 +492,32 @@ void InitDeviceModelDb::insert_attribute(const VariableAttribute& attribute, con
     if (insert_attributes_statement->step() != SQLITE_DONE) {
         throw InitDeviceModelDbError("Could not insert attribute: " + std::string(this->database->get_error_message()));
     }
+
+    const int64_t attribute_id = this->database->get_last_inserted_rowid();
+
+    if (attribute.value.has_value() ||
+        (attribute.type.has_value() && (attribute.type.value() == AttributeEnum::Actual) &&
+         default_actual_value.has_value())) {
```
Comment (Pietfried) [inline lib/ocpp/v201/init_device_model_db.cpp:500]: You could put this into a separate function so this reads a little better. You could also reusue it within L604

**Review by Pietfried — APPROVED** [review]
Looks nice. Just added a couple of suggestions. Make sure to update the dependency in https://github.com/EVerest/everest-core/pull/822 once this is merged

File: lib/ocpp/v201/ocpp_types.cpp [hunk 2]
```diff
@@ -1866,9 +1866,28 @@ void from_json(const json& j, VariableAttribute& k) {
     if (j.contains("type")) {
         k.type.emplace(conversions::string_to_attribute_enum(j.at("type")));
     }
+
+    // TODO mz change this in the generator (if everyone agrees)!!
```
[the remark this answers is not in the document]
Comment (maaikez) [reply lib/ocpp/v201/ocpp_types.cpp:1870]: So you don't generate it anymore?
(I just want to be sure it is not removed when the file is regenerated)

**Comment by acnixvkh** [comment]
@maaikez, I encountered some difficulty adapting my configuration when changes from this pull request appeared in my copy of EVerest. Although it is working again, I am unsure if my approach is correct and would appreciate your confirmation.

**Scenario:** I want to change a value defined in the standardized schemas or define a value for a variable that has no default value in the standardized schemas. To do this, I create a new schema in the sub-directory `component_config/custom/`. This new schema repeats much of the code from the original schema in `component_config/standardized/` with only the `"value"` element added or changed.

**Example:** To set `ContractCertificateInstallationEnabled` from `component_config/standardized/ISO15118Ctrlr.json` to `true`, I create `component_config/custom/config.json` with the following content:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "name": "ISO15118Ctrlr",
  "properties": {
    "ContractCertificateInstallationEnabled": {
      "variable_name": "ContractCertificateInstallationEnabled",
      "characteristics": {
        "supportsMonitoring": true,
        "dataType": "boolean"
      },
      "attributes": [
        {
          "type": "Actual",
          "mutability": "ReadWrite",
          "value": true
        }
      ],
      "description": "If this variable is true, then ISO 15118 contract certificate installation/update as described by use case M01 - Certificate installation EV and M02 - Certificate Update EV is enabled. If this variable is false, then ISO 15118 contract certificate installation/update as described by use case M01 - Certificate installation EV and M02 - Certificate Update EV is disabled.",
      "type": "boolean"
    }
  }
}
```

**Question:** Is this the correct approach to adapt configurations in libocpp, or is there a more efficient method to avoid repeating large portions of code?

**Comment by maaikez** [comment]
@acnixvkh Ah never thought of this way. No you should just change the ISO15118Ctrlr.json and set the correct value there. Although this might just work, but I am not sure. 

The default values are confusing, I will remove them in the future.

**Pull request merged**